src/error.rs

```diff
@@ -6,73 +6,108 @@ use std::error::Error as StdError;
 #[cfg(feature = "std")]
 use self::super::TriviallyTransmutable;
 use core::fmt;
+#[cfg(feature = "std")]
+use core::marker::PhantomData;
 
 
 /// A transmutation error. This type describes possible errors originating
-/// from operations in this crate.
+/// from operations in this crate. The two type parameters, when defined,
+/// represent the source element type and the target element type
+/// respectively.
 ///
 /// # Examples
 ///
 /// ```
 /// # use safe_transmute::{ErrorReason, Error, transmute_bool_pedantic};
-/// # unsafe {
 /// assert_eq!(transmute_bool_pedantic(&[0x05]), Err(Error::InvalidValue));
-/// # }
 /// ```
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
-pub enum Error {
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub enum Error<T = (), U = ()> {
```

Comment:
I'd name the types `From`/`To` or `Source`/`Target` or sth, rather than `T`/`U`, esp. when `T` is used as the *T*arget type elsewhere

Reply:
I am not fond of non-short type parameters. `From` might also be confused with the trait. I agree with the `T`/`U` pair being inconsistent with the rest of the code however, so I'll rename everything to `S`/`T` for the time being.